checks/containers/container_engine/check_cuda_nbody.py

```diff
@@ -21,8 +21,11 @@ class CudaNBodyCheckCE(rfm.RunOnlyRegressionTest, ContainerEngineMixin):
     sourcesdir = None
     num_tasks = 1
     num_tasks_per_node = 1
-    executable = '/cuda-samples/nbody'
-    container_image = 'nvcr.io#nvidia/k8s/cuda-sample:nbody-cuda11.7.1'
+    executable = 'nbody'
+    container_image = 'jfrog.svc.cscs.ch#reframe-oci/cuda_samples:nbody-12.3'
+    env_vars = {
+        'NVIDIA_DISABLE_REQUIRE': 1
+    }
     reference = {
         '*': {
             'gflops': (28000., -0.05, None, 'Gflop/s')
```

Comment:
This is kind of tricky because the number of bodies that you require has to be adjusted accordingly so it needs some runs before you find the best performance.